- Documentation for Indi component

shaunmaharaj opened the pull request:
Description:
<!--A brief description of changes. Things to include: WIP? Dependent PR's opened against other tickets? -->
- Add documentation for Indi Component
(Previously merged in #140 )

Linting:
<!--Have you validated that no linting errors are introduced? -->
- [x] No linting errors

Tests:
<!--Have tests been run locally and passed? If manual tests run, explain what was run below-->
- [ ] Smoke tests (mvn clean install -Dcucumber.options="--tags @smoketest" from test)
- [ ] Manual tests

Documentation:
<!--Are documentation updates required? Include any mention of updates required below if necessary-->
- [x] Requires documentation updates

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+* `configuration`: The structured object consisting of the configurations to provide to the Indi components. These are defined in `ep.config.json` in the `indi` element.
+* `keywords`: A string of the keywords for the product in which the Indi component is displayed for.
+
+Static strings for the Indi component have been localized in the Reference Storefront and are overrided on a per-usage basis for the Indi component. This will ensure the component displays the correct text in the storefront based on the locale which a customer chooses during their shopping flow.
```
```suggestion
Static strings for the Indi component are localized in the Reference Storefront and are overridden whenever the Indi component is used. With this setting, the component displays the storefront in the locale that a customer chooses.
```

sminnath replied:
Why is this section here? Why are we suddenly talking about Static strings here when in the next section we are going to explain `configuration`?

shaunmaharaj replied:
I wanted to call out that we provide localization options for the strings a developer would normally pass to the Indi library to display the component. This allows for other languages to be used for the component dynamically rather than hardcoding the strings through configurations.

sminnath replied:
Do you want to have a heading here? @shaunmaharaj

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+* `keywords`: A string of the keywords for the product in which the Indi component is displayed for.
+
+Static strings for the Indi component have been localized in the Reference Storefront and are overrided on a per-usage basis for the Indi component. This will ensure the component displays the correct text in the storefront based on the locale which a customer chooses during their shopping flow.
+
```
Are you specifically talking about the configuration in the  `ep.config.json` in the `indi` parameter? 
If yes this should be a sub section with the title 
**Configuration Settings for Indi in the ep.config.json file**

shaunmaharaj replied:
Nope, I'm referring to the files in `src/localization`.
Both of these two:
`src/localization/en-CA.json`
`src/localization/fr-FR.json`

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+
+Static strings for the Indi component have been localized in the Reference Storefront and are overrided on a per-usage basis for the Indi component. This will ensure the component displays the correct text in the storefront based on the locale which a customer chooses during their shopping flow.
+
+Configuration properties for Indi component's requirements are:
```
Following are the configuration settings for the `indi` parameter in the `ep.config.json` file:

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+Static strings for the Indi component have been localized in the Reference Storefront and are overrided on a per-usage basis for the Indi component. This will ensure the component displays the correct text in the storefront based on the locale which a customer chooses during their shopping flow.
+
+Configuration properties for Indi component's requirements are:
+* `indi.enable`: Enable the integration component for Indi. More information may be found available here [https://indi.com/](https://indi.com/).
```
```suggestion
* `indi.enable`: The parameter that enables the component for the integration with Indi. For more information about the required configurations, see the company [website](https://indi.com/).
```

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+Configuration properties for Indi component's requirements are:
+* `indi.enable`: Enable the integration component for Indi. More information may be found available here [https://indi.com/](https://indi.com/).
+* `indi.carousel`: Configurations for the Indi carousel component.
+* `indi.carousel.apikey`: The apikey used to connect the carousel to Indi.
```
```suggestion
* `indi.carousel.apikey`: The _apikey_ that connects a carousel to Indi.
```
use the carousel if it is for the above mentioned one

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+
+Configuration properties for Indi component's requirements are:
+* `indi.enable`: Enable the integration component for Indi. More information may be found available here [https://indi.com/](https://indi.com/).
+* `indi.carousel`: Configurations for the Indi carousel component.
```
The configurations for the Indi carousel component.

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+* `indi.enable`: Enable the integration component for Indi. More information may be found available here [https://indi.com/](https://indi.com/).
+* `indi.carousel`: Configurations for the Indi carousel component.
+* `indi.carousel.apikey`: The apikey used to connect the carousel to Indi.
+* `indi.carousel.id`: The apikey used to connect the carousel to Indi.
```
this doesnt look correct. You mean id?

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+* `indi.carousel`: Configurations for the Indi carousel component.
+* `indi.carousel.apikey`: The apikey used to connect the carousel to Indi.
+* `indi.carousel.id`: The apikey used to connect the carousel to Indi.
+* `indi.carousel.size`: Used when theming the carousel displaying content from Indi.
```
```suggestion
* `indi.carousel.size`: The parameter used to theme the carousel that displays the content from Indi.
```

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+* `indi.productReview.thumbnail_url`: The thumbnail provided by Indi for the submit button.
+* `indi.brandAmbassador`: Configurations for the Indi brand ambassador component.
+* `indi.brandAmbassador.submit_button_url`: The URL provided by Indi which directs to the action performed when the submit button is clicked on a brand ambassador signup.
+* `indi.brandAmbassador.thumbnail_url`: The thumbnail provided by Indi for the submit button.
```
Can we have a table with the following details:

Functionality|     Parameter |            Description |  >> **Table header**

**Integration**

**Theming**          |  List all parameters from (101 to 108)|    Description for 101
                                                                                                    ...


**Product View** .  |  (line 109 - 111)|

**Brand ambassador** |(line 112-114) |

shaunmaharaj replied:
Adding in latest commit

asmurthwaite commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+The Indi component available is a stand-alone component created solely to fetch the `indi-embed` libraries, then display the corresponding elements that were passed into the component as a structured list.
+
+The required parameters of this component are as follows:
+* `render`: A list of the elements to display from Indi. Example: `render={['carousel', 'brand', 'product']}`. Note the correct configurations must be provided for the element to render correctly.
```
I'm just looking at the comments, but one question I have as someone new (and maybe if I saw the whole page, this would make sense) but in the note, you mention correct configurations are required, is there an example of some configurations?  If I were to do it, I would want a guide.

shaunmaharaj replied:
Example provided in `ep.config.json`

shaunmaharaj commented on documentation/practices.md:
```diff
@@ -80,5 +80,37 @@ Configuration properties for content URLs are:
+The required parameters of this component are as follows:
+* `render`: A list of the elements to display from Indi. Example: `render={['carousel', 'brand', 'product']}`. Note the correct configurations must be provided for the element to render correctly.
+* `configuration`: The structured object consisting of the configurations to provide to the Indi components. These are defined in `ep.config.json` in the `indi` element.
+* `keywords`: A string of the keywords for the product in which the Indi component is displayed for.
```
@sminnath It's the former - `for the product for which the indi component is displayed`

I'll update for that 👍

shaunmaharaj commented:
@sminnath @asmurthwaite I've addressed all your comments. Thank you for the review!
Please mark as approved if there are no further comments then I'll merge.

Should this page get split up in the future? Right now there's a lot of material on a single page and potentially it can be broken up similar to how our Tutorials are setup.

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+
+**Brand Loyalty Integration for Indi**
+
+The React PWA Reference Storefront is pre-configured with a custom component for Indi integration. This component interacts with various widgets that Indi displays. New widgets may be added in the component as made available. For more information on Indi, see the company [website](https://indi.com/).
```
When Indi adds new widgets, you can update the component with the new widget.

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+
+This component requires the following parameters:
+* `render`: The list of the elements to display from Indi. For example, `render={['carousel', 'brand', 'product']}`.
+Note:  Provide the correct configurations to render the element  correctly.
```
extra space after element

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+This component requires the following parameters:
+* `render`: The list of the elements to display from Indi. For example, `render={['carousel', 'brand', 'product']}`.
+Note:  Provide the correct configurations to render the element  correctly.
+* `configuration`: The structured object that consists of the configuration required for the Indi components. These configurations are defined in the `indi` parameter in `react-pwa-reference-storefront⁩/ep.config.json' file.
```
```suggestion
* `configuration`: The structured object that consists of the configuration required for the Indi components. These configurations are defined in the `indi` parameter in the `react-pwa-reference-storefront⁩/ep.config.json' file.
```

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+The Indi component is a stand-alone component to fetch only the `indi-embed` libraries and display the elements corresponding to the libraries. These elements are passed to the component as a structured list.
+
+This component requires the following parameters:
+* `render`: The list of the elements to display from Indi. For example, `render={['carousel', 'brand', 'product']}`.
```
Is there a specific format for this parameter? or should we just say. 
For example, see the `ep.config.json' file settings, `render={['carousel', 'brand', 'product']}`.

shaunmaharaj replied:
Yep there's a format. I'll need to keep the example

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+* `configuration`: The structured object that consists of the configuration required for the Indi components. These configurations are defined in the `indi` parameter in `react-pwa-reference-storefront⁩/ep.config.json' file.
+* `keywords`: A string of the keywords for the product in which the Indi component is displayed.
+
+Static strings for the Indi component are localized in the Reference Storefront and are overridden whenever the Indi component is used. With this setting, the component displays the storefront in the locale that a customer chooses. The localized strings for the Indi component can be found in the corresponding language files within `src/localization`, for strings with the prefix `indi-`.
```
```suggestion
Static strings for the Indi component are localized in the Reference Storefront and are overridden whenever the component is used. With this setting, the component displays the storefront in the locale that a customer chooses. The localized strings for the Indi component are available in the corresponding language files in the `src/localization` directory. This setting is only for the strings with the `indi-`prefix.
```

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+| |`indi.carousel.show_views`| The parameter used to theme the carousel that displays the content from Indi.|
+| |`indi.carousel.show_likes`| The parameter used to theme the carousel that displays the content from Indi.|
+| |`indi.carousel.show_buzz`| The parameter used to theme the carousel that displays the content from Indi.|
+| |`indi.carousel.animate`| The parameter used to theme the carousel that displays the content from Indi.|
```
Can we merge all rows with one `The parameter used to theme the carousel that displays the content from Indi.`

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+| |`indi.carousel.show_likes`| The parameter used to theme the carousel that displays the content from Indi.|
+| |`indi.carousel.show_buzz`| The parameter used to theme the carousel that displays the content from Indi.|
+| |`indi.carousel.animate`| The parameter used to theme the carousel that displays the content from Indi.|
+| **Product Review**|`indi.productReview`| Configurations for the Indi product review component.|
```
The configurations

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+| |`indi.carousel.show_buzz`| The parameter used to theme the carousel that displays the content from Indi.|
+| |`indi.carousel.animate`| The parameter used to theme the carousel that displays the content from Indi.|
+| **Product Review**|`indi.productReview`| Configurations for the Indi product review component.|
+| |`indi.productReview.submit_button_url`| The URL provided by Indi which directs to the action performed when the submit button is clicked on a product review.|
```
```suggestion
| |`indi.productReview.submit_button_url`| The URL of the page to which Indi redirects when a brand ambassador submits a product review.

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+| **Product Review**|`indi.productReview`| Configurations for the Indi product review component.|
+| |`indi.productReview.submit_button_url`| The URL provided by Indi which directs to the action performed when the submit button is clicked on a product review.|
+| |`indi.productReview.thumbnail_url`| The thumbnail provided by Indi for the submit button.|
+| **Brand Ambassador**|`indi.brandAmbassador`| Configurations for the Indi brand ambassador component.|
```
```suggestion
| **Brand Ambassador**|`indi.brandAmbassador`| The configurations for the Indi brand ambassador component.|
```

sminnath replied:
Is the component name brand ambassador?

shaunmaharaj replied:
The config for it is

sminnath replied:
configuration for brand ambassador component in Indi, right?

shaunmaharaj replied:
yes, that's correct

sminnath commented on documentation/practices.md:
```diff
@@ -80,5 +80,43 @@ Configuration properties for content URLs are:
+| |`indi.productReview.submit_button_url`| The URL provided by Indi which directs to the action performed when the submit button is clicked on a product review.|
+| |`indi.productReview.thumbnail_url`| The thumbnail provided by Indi for the submit button.|
+| **Brand Ambassador**|`indi.brandAmbassador`| Configurations for the Indi brand ambassador component.|
+| |`indi.brandAmbassador.submit_button_url`| The URL provided by Indi which directs to the action performed when the submit button is clicked on a brand ambassador signup.|
```
```suggestion
| |`indi.brandAmbassador.submit_button_url`| The URL to which Indi redirects when a brand ambassador signs up.|
```
